### PR TITLE
Added site detection for cloudflare and updated noData page to mention cloudflare

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:
Added site detection for cloudflare and updated noData page to mention cloudflare
Fixes: PG-2726
Cherry picked [Cloudflare](https://github.com/matomo-org/matomo/commit/8917e2680117d4073ed77f8ac1bbdb360e8914d8) changes from 5.x-dev.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/SitesManager/SitesManager.php [hunk 1]
```diff
@@ -506,12 +506,21 @@ public function getClientSideTranslationKeys(&$translationKeys)
         $translationKeys[] = "SitesManager_JsTrackingTagHelp";
         $translationKeys[] = "SitesManager_SiteWithoutDataSinglePageApplication";
         $translationKeys[] = "SitesManager_SiteWithoutDataSinglePageApplicationDescription";
+        $translationKeys[] = 'SitesManager_SiteWithoutDataTitle';
+        $translationKeys[] = 'SitesManager_SiteWithoutDataDescription';
+        $translationKeys[] = 'SitesManager_SiteWithoutDataMessageDisappears';
+        $translationKeys[] = 'SitesManager_SiteWithoutDataChoosePreferredWay';
+        $translationKeys[] = 'SitesManager_DetectingYourSite';
+        $translationKeys[] = 'SitesManager_SiteWithoutDataIgnoreMessage';
```
Comment (sgiehl) [inline plugins/SitesManager/SitesManager.php]: Are those needed. They weren't added in the original PR.

File: plugins/SitesManager/SitesManager.php [hunk 2]
```diff
@@ -506,12 +506,21 @@ public function getClientSideTranslationKeys(&$translationKeys)
...
+        $translationKeys[] = "SitesManager_SiteWithoutDataCloudflare";
+        $translationKeys[] = "SitesManager_SiteWithoutDataCloudflareDescription";
         $translationKeys[] = "SitesManager_GlobalListExcludedReferrers";
         $translationKeys[] = "SitesManager_GlobalListExcludedReferrersDesc";
         $translationKeys[] = "SitesManager_ExcludedReferrers";
         $translationKeys[] = "SitesManager_ExcludedReferrersHelp";
         $translationKeys[] = "SitesManager_ExcludedReferrersHelpDetails";
         $translationKeys[] = "SitesManager_ExcludedReferrersHelpExamples";
         $translationKeys[] = "SitesManager_ExcludedReferrersHelpSubDomains";
+        $translationKeys[] = 'Goals_Optional';
```
Comment (sgiehl) [inline plugins/SitesManager/SitesManager.php]: This one as well.

File: plugins/SitesManager/SitesManager.php (hunk 1 above)
Comment (AltamashShaikh) [reply plugins/SitesManager/SitesManager.php]: Removed, I didn't resolve merge conflicts correctly

File: plugins/SitesManager/SitesManager.php (hunk 2 above)
Comment (AltamashShaikh) [reply plugins/SitesManager/SitesManager.php]: Removed, I didn't resolve merge conflicts correctly

**Review by sgiehl — APPROVED** [review]
Seems to work as expected.

**Pull request merged**